bundles/com.espressif.idf.launch.serial.core/src/com/espressif/idf/launch/serial/internal/SerialFlashLaunchConfigDelegate.java

```diff
@@ -190,7 +192,7 @@ protected void launchInternal(ILaunchConfiguration configuration, String mode, I
 			buffer.append('=').append(entry.getValue());
 			strings.add(buffer.toString());
 		}
-
+		Logger.log(String.format("flash command: %s", String.join(" ", commands))); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
can be replaced with the Messages class, but we are already using Messages from org.eclipse.cdt.debug.internal.core package in this class, which btw provides confusing error messages in our case. Maybe it is worth to do refactoring to provide our own Messages class, which will make more sense in our use cases

Reply:
That make sense, we can create a Message class inside the plugin to avoid internal reference issues

Reply:
Do we need Messages class for logger? We need to be careful when utilizing the Messages in our Logs as they are language dependent and incase of a different language the logs may switch to that language making it difficult when we try to debug any issue in future with logs.